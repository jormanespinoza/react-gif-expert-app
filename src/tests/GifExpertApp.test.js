import React from 'react'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Testing GifExpertApp component', () => {
    test('should render <GifExpertApp /> properly', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show a list of categories', () => {
        const categories = ['Dragon Ball', 'Caballeros del Zodiaco'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
