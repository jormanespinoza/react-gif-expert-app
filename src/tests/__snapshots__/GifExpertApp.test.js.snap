// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp component should render <GifExpertApp /> properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
